superset/connectors/sqla/models.py

```diff
@@ -1901,9 +1903,7 @@ def update_table(  # pylint: disable=unused-argument
         session.execute(update(SqlaTable).where(SqlaTable.id == target.table.id))
 
         dataset = (
-            session.query(NewDataset)
-            .filter_by(sqlatable_id=target.table.id)
-            .one_or_none()
+            session.query(NewDataset).filter_by(uuid=target.table.uuid).one_or_none()
```

Comment:
We now link old and new datasets by uuid instead of a `sqlatable_id` field. Note since `NewDataset` is designed to be a 1:1 replacement for SqlaTable, it should be okay for them to share the same uuid. However, `NewTable` is a totally new entity type, so they should have separate uuids. Therefore a `sqlatable_id` is added to `NewTable` instead.